src/option_parser.c

```diff
@@ -250,22 +233,46 @@ int load_ini_file(FILE * fp)
 
                 *equal = '\0';
                 char *key = g_strstrip(start);
-                char *value = g_strstrip(equal + 1);
-
-                char *quote = strchr(value, '"');
-                if (quote) {
-                        char *closing_quote = strchr(quote + 1, '"');
-                        if (!closing_quote) {
-                                fprintf(stderr,
-                                     "Warning: invalid config file at line %d\n",
-                                     line_num);
-                                fprintf(stderr, "Missing '\"'\n");
-                                continue;
+                char *unparsed = equal + 1;
+                char *value = unparsed;
+
+                bool in_quote = false;
+                while ((unparsed = strpbrk(unparsed, "\"\\#;")) != NULL) {
+                        switch (*unparsed) {
+                        case '"':
+                                memmove(unparsed, unparsed + 1, strlen(unparsed));
+                                in_quote = !in_quote;
+                                break;
+                        case '\\':
+                                switch(unparsed[1]) {
+                                case '\\':
+                                case '"':
+                                        memmove(unparsed, unparsed + 1, strlen(unparsed));
+                                        unparsed++;
+                                        break;
+                                default:
+                                        // Unrecognized backslash sequence;
+                                        // skip over it without removing the backslash.
+                                        // Consider issuing an error or warning here instead.
+                                        unparsed += 2;
```

Comment:
If the backslash is at the end of string, this will increment `unparsed` too much and leave next `strpbrk` undefined.

Add `-fsanitize=address` to `CFLAGS`, and use this config to reproduce:
```
[global]
    unused=\
```
Note that there is no trailing new line, i.e.
```
$ xxd /tmp/dunstrc 
00000000: 5b67 6c6f 6261 6c5d 0a20 2020 2075 6e75  [global].    unu
00000010: 7365 643d 5c                             sed=\
```